packages/@lwc/engine-core/src/framework/context.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ContextEventName, getContextKeys } from '@lwc/shared';
+import type { Signal } from '@lwc/signals';
+
+export type ContextProvidedCallback = (contextSignal: Signal<unknown>) => void;
+export class ContextRequestEvent extends CustomEvent<{
+    key: symbol;
+    contextVariety: unknown;
+    callback: ContextProvidedCallback;
+}> {
+    constructor(detail: { contextVariety: unknown; callback: ContextProvidedCallback }) {
+        super(ContextEventName, {
+            bubbles: true,
+            composed: true,
+            detail: { ...detail, key: getContextKeys().contextEventKey },
+        });
+    }
+}
```

Comment:
Have you looked at the [Web Components Community Group proposal for a "context" event](https://github.com/webcomponents-cg/community-protocols/blob/4001efa99417850f7c023727be9ddc8875364672/proposals/context.md#the-context-request-event)? This is an attempt to standardize the concept of passing context between ancestor/descendant components across web component frameworks.

The advantage of using this protocol is that, hypothetically, a non-LWC framework like Lit could use the same protocol and be interoperable with LWC in the same DOM tree.